src/Sharpliner/AzureDevOps/ConditionedExpressions/ConditionBuilder.cs

```diff
@@ -147,10 +245,16 @@ public Condition NotIn(string needle, params string[] haystack)
     public Condition<T> ContainsValue(string needle, params string[] haystack)
         => new ContainsValueCondition<T>(needle, haystack);
 
-    public Condition<T> IsBranch(string branchName)
+    public Condition<T> IsBranch(StringRuntimeExpression branchName)
         => Link(new BranchCondition<T>(branchName, true));
 
-    public Condition<T> IsNotBranch(string branchName)
+    public Condition<T> IsBranch(IRuntimeExpression branchName)
+        => Link(new BranchCondition<T>(branchName, true));
+
+    public Condition<T> IsNotBranch(StringRuntimeExpression branchName)
+        => Link(new BranchCondition<T>(branchName, false));
+
+    public Condition<T> IsNotBranch(IRuntimeExpression branchName)
         => Link(new BranchCondition<T>(branchName, false));
 
     public Condition<T> Greater(string first, string second)
```

Comment:
`Greater` and `Less` should ideally also work with the new expressions

Reply:
Done